enumerables.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+    def my_each
+        for i in 0...self.length do
+            yield(self[i])
+        end
+    end
+    
+    def my_each_with_index
+        for i in 0...self.length do
+            yield(self[i], i)
+        end
+    end
+
+    def my_select
+        a = []
+        my_each do |n|
+            if yield(n)
+                a.push(n)
+            end
+        end
+        a
+    end
+
+    def my_all?
+        my_each do |n|
+            if !yield(n)
+                return false
+            end
+        end
+        true
+    end
+
+    def my_none?
+        my_each do |n|
+            if yield(n)
+                return false
+            end
+        end
+        true
+    end
+
```

Comment:
- [ ] Same problem not handling classes, regex and patterns is present on my_none? also it should return false if the block ever returns true for all elements::range.